qiskit/algorithms/minimum_eigen_solvers/vqe.py

```diff
@@ -414,103 +438,114 @@ def compute_minimum_eigenvalue(
-                    bind_params=self._ansatz_params,
-                    backend=self._quantum_instance,
-                )
-        if not self._expect_op:
-            self._expect_op = self.construct_expectation(self._ansatz_params, operator)
-        vqresult = self.find_minimum(
-            initial_point=self.initial_point,
-            ansatz=self.ansatz,
-            cost_fn=self._energy_evaluation,
-            gradient_fn=self._gradient,
-            optimizer=self.optimizer,
+        start_time = time()
+        opt_params, opt_value, nfev = self.optimizer.optimize(
+            num_vars=len(initial_point),
+            objective_function=energy_evaluation,
+            gradient_function=gradient,
+            variable_bounds=bounds,
+            initial_point=initial_point,
         )
+        eval_time = time() - start_time
 
-        self._ret = VQEResult()
-        self._ret.combine(vqresult)
+        result = VQEResult()
+        result.optimal_point = opt_params
+        result.optimal_parameters = dict(zip(self._ansatz_params, opt_params))
+        result.optimal_value = opt_value
+        result.cost_function_evals = nfev
+        result.optimizer_time = eval_time
+        result.eigenvalue = opt_value + 0j
+        result.eigenstate = self._get_eigenstate(result.optimal_parameters)
 
-        if vqresult.optimizer_evals is not None and self._eval_count >= vqresult.optimizer_evals:
-            self._eval_count = vqresult.optimizer_evals
-        self._eval_time = vqresult.optimizer_time
         logger.info(
             "Optimization complete in %s seconds.\nFound opt_params %s in %s evals",
-            self._eval_time,
-            vqresult.optimal_point,
+            eval_time,
+            result.optimal_point,
             self._eval_count,
         )
 
-        self._ret.eigenvalue = vqresult.optimal_value + 0j
-        self._ret.eigenstate = self.get_optimal_vector()
-        self._ret.eigenvalue = self.get_optimal_cost()
-        if aux_operators:
-            self._eval_aux_ops(aux_operators)
-            self._ret.aux_operator_eigenvalues = self._ret.aux_operator_eigenvalues[0]
+        # TODO delete as soon as get_optimal_vector etc are removed
+        self._ret = result
 
-        self._ret.cost_function_evals = self._eval_count
+        if aux_operators is not None:
+            aux_values = self._eval_aux_ops(opt_params, aux_operators)
+            result.aux_operator_eigenvalues = aux_values[0]
 
-        return self._ret
+        return result
 
-    def _energy_evaluation(
-        self, parameters: Union[List[float], np.ndarray]
-    ) -> Union[float, List[float]]:
+    def get_energy_evaluation(
+        self, operator: OperatorBase
+    ) -> Callable[[np.ndarray], Union[float, List[float]]]:
         """Evaluate energy at given parameters for the ansatz.
 
         This is the objective function to be passed to the optimizer that is used for evaluation.
```

Comment:
I think this description could be updated. This is no longer the objective function itself but a method which will return the objective function - at least this is how it looks to me